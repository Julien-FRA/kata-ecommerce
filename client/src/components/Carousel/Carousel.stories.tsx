import { fn } from "@storybook/test";

import { Carousel } from ".";

export default {
  title: "Carousel",
  component: "Carousel",
  args: {
    children: "Carousel",
  },
  parameters: {
    docs: {
      toc: true, // ðŸ‘ˆ Enables the table of contents
    },
  },
};

export const Default = {
  render: () => <Carousel />,
};
