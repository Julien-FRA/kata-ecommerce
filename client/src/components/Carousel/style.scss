.carousel-container {
  width: 50%;
  display: flex;
  flex-direction: row-reverse;
  align-items: flex-start;

  @media (max-width: 996px) {
    width: 100%;
  }

  img {
    width: 100%;
    height: auto;
    object-fit: cover;
  }

  .image-container {
    position: relative;
    width: 80%;

    figure {
      &:not(.show) {
        position: absolute;
        top: 0;
        right: 0;
      }
    }
  }

  .button-container {
    width: 20%;
    max-height: 412px;
    overflow: hidden;

    @media (max-width: 996px) {
      max-height: 562px;
    }

    button {
      padding: 0;
      border: none;
      cursor: pointer;
    }
  }
}
