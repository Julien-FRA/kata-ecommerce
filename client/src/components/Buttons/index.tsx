import React from "react";
import "./style.scss";

interface ButtonProps {
  type?: "button" | "submit" | "reset";
  href?: string;
  variant: "red" | "white" | "black";
  size: "small" | "large";
  content: string;
  onClick?: () => void;
}

export const Button = ({
  type,
  href,
  variant,
  size,
  content,
  onClick,
}: ButtonProps) => {
  return href ? (
    <a
      href={href}
      className={`button-container ${variant} ${size}`}
      onClick={onClick}
    >
      {content}
    </a>
  ) : (
    <button
      type={type}
      className={`button-container ${variant} ${size}`}
      onClick={onClick}
    >
      {content}
    </button>
  );
};
