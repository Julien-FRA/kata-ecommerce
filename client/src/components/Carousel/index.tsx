import React, { useState } from "react";
import "./style.scss";
import product from "../../assets/images/product_1.webp";

export const Carousel = () => {
  const [slide, setSlide] = useState<number>(0);

  const getSlide = (value: number) => {
    setSlide(value);
  };

  const showSlide = () => {};

  return (
    <div className="carousel-container">
      <div className="image-container">
        <figure className="show">
          <img src={product} alt="product" />
        </figure>
        <figure>
          <img src={product} alt="product" />
        </figure>
        <figure>
          <img src={product} alt="product" />
        </figure>
      </div>
      <div className="button-container">
        <button key={0} onClick={() => getSlide(0)}>
          <figure>
            <img src={product} alt="product" />
          </figure>
        </button>
        <button key={1} onClick={() => getSlide(1)}>
          <figure>
            <img src={product} alt="product" />
          </figure>
        </button>
        <button key={2} onClick={() => getSlide(2)}>
          <figure>
            <img src={product} alt="product" />
          </figure>
        </button>
      </div>
    </div>
  );
};
