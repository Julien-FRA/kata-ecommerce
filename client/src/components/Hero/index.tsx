import React from "react";
import logo from "../../assets/images/logo_kokoro_3d.compress.webp";
import "./style.scss";

export const Hero = () => {
  return (
    <div className="hero">
      <div className="wrapper">
        <div className="left-content">
          <h1>Decouvre notre nouvelle collection de 2025</h1>
          <h2>
            Made by <span>Kokoro</span>
          </h2>
          <a href="/">DÃ©couvrir</a>
        </div>
        <div className="right-content">
          <img src={logo} alt="logo" />
        </div>
      </div>
    </div>
  );
};
