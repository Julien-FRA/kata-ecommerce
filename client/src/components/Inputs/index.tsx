import React, { useState } from "react";
import "./style.scss";

interface InputProps {
  type: "text" | "email" | "password";
  name: string;
  id: string;
  placeholder: string;
  disabled?: boolean;
}

export const Input = ({
  type,
  name,
  id,
  placeholder,
  disabled,
}: InputProps) => {
  const [value, setValue] = useState<string>("");

  return (
    <>
      {disabled ? (
        <input
          className="input disable-input"
          type={type}
          name={name}
          id={id}
          placeholder={placeholder}
          value={value}
          onChange={(e) => setValue(e.target.value)}
          disabled={disabled}
        />
      ) : (
        <input
          className="input"
          type={type}
          name={name}
          id={id}
          placeholder={placeholder}
          value={value}
          onChange={(e) => setValue(e.target.value)}
        />
      )}
    </>
  );
};
