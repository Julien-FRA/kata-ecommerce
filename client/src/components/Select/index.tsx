import React from "react";
import "./style.scss";
import { Button } from "../Buttons";

interface SelectProps {
  name: string;
  price: number;
  size: "XS" | "S" | "M" | "L" | "XL";
  color: string;
}

export const Select = ({ name, price, size, color }: SelectProps) => {
  return (
    <div className="select-container">
      <h2>{name}</h2>
      <p className="price">{price}â‚¬</p>
      <form>
        <div className="option">
          <label className="color p">Taille</label>
          <select name="" id="">
            <option value="marron">{size}</option>
            <option value="bleu">{size}</option>
            <option value="bleu">{size}</option>
          </select>
        </div>
        <div className="option">
          <label className="color p">Couleurs</label>
          <select name="" id="">
            <option value="marron">{color}</option>
            <option value="bleu">{color}</option>
          </select>
        </div>
        <Button
          type={"submit"}
          variant={"red"}
          size={"large"}
          content={"Ajouter au panier"}
        />
      </form>
    </div>
  );
};
