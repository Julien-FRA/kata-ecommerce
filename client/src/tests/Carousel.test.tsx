import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Carousel } from "../components/Carousel";

describe("<Carousel />", () => {
  test(`Basic test`, () => {
    // Génaration du DOM virtuel
    render(<Carousel />);

    const image = screen.getAllByRole("img");

    // Test d'image sur la valeur affiché en premier sur le carousel
    expect(image[0]).toHaveAttribute("src", "product_1.webp");
    expect(image[0]).toHaveAttribute("alt", "product");
  });

  test(`Click du user sur le button de l'image déja affiché`, async () => {
    // Génaration du DOM virtuel
    render(<Carousel />);

    const image = screen.getAllByRole("img");

    const button = screen.getAllByRole("button");

    await userEvent.click(button[0]);
    expect(image[0]).toHaveAttribute("src", "product_1.webp");
  });

  test(`Changement d'image au click du user`, async () => {
    // Génaration du DOM virtuel
    render(<Carousel />);

    const image = screen.getAllByRole("img");

    const button = screen.getAllByRole("button");

    await userEvent.click(button[1]);
    expect(image[0]).toHaveAttribute("src", "product_2.webp");
  });
});
