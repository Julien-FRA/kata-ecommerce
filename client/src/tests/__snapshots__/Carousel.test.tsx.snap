// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Carousel /> Snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="carousel-container"
      >
        <div
          class="image-container"
        >
          <figure
            class="show"
          >
            <img
              alt="product"
              src="product_1.webp"
            />
          </figure>
        </div>
        <div
          class="button-container"
        >
          <button>
            <figure>
              <img
                alt="product"
                src="product_1.webp"
              />
            </figure>
          </button>
          <button>
            <figure>
              <img
                alt="product"
                src="product_2.webp"
              />
            </figure>
          </button>
          <button>
            <figure>
              <img
                alt="product"
                src="product_3.webp"
              />
            </figure>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="carousel-container"
    >
      <div
        class="image-container"
      >
        <figure
          class="show"
        >
          <img
            alt="product"
            src="product_1.webp"
          />
        </figure>
      </div>
      <div
        class="button-container"
      >
        <button>
          <figure>
            <img
              alt="product"
              src="product_1.webp"
            />
          </figure>
        </button>
        <button>
          <figure>
            <img
              alt="product"
              src="product_2.webp"
            />
          </figure>
        </button>
        <button>
          <figure>
            <img
              alt="product"
              src="product_3.webp"
            />
          </figure>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
